[Bug]: mvn clean package v2 FAILED due to splunk.jfrog.io repo

### Related Template(s)

MySQL_to_BigQuery

### Template Version

v2

### What happened?

Following the provided documentation [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/mysql-to-googlecloud/README_MySQL_to_BigQuery.md) to build MySQL_to_BigQuery (v2), I keep getting the following error :


### Relevant log output

```shell
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.145 s
[INFO] Finished at: 2024-05-11T10:58:22+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project mysql-to-googlecloud: Could not resolve dependencies for project com.google.cloud.teleport.v2:mysql-to-googlecloud:jar:1.0-SNAPSHOT: Failed to collect dependencies at com.google.cloud.teleport.v2:jdbc-to-googlecloud:jar:1.0-SNAPSHOT: Failed to read artifact descriptor for com.google.cloud.teleport.v2:jdbc-to-googlecloud:jar:1.0-SNAPSHOT: The following artifacts could not be resolved: com.google.cloud.teleport.v2:jdbc-to-googlecloud:pom:1.0-SNAPSHOT (absent): Could not transfer artifact com.google.cloud.teleport.v2:jdbc-to-googlecloud:pom:1.0-SNAPSHOT from/to splunk-artifactory (https://splunk.jfrog.io/splunk/ext-releases-local): status code: 409, reason phrase:  (409) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

I have a similar issue just trying to `mvn clean compile` from the top level.

```
[ERROR] The following artifacts could not be resolved: io.confluent:kafka-schema-registry-client:jar:7.6.0 (present, but unavailable), io.confluent:kafka-avro-serializer:jar:7.6.0 (present, but unavailable), io.confluent:common-utils:jar:7.6.0 (present, but unavailable), io.confluent:kafka-schema-serializer:jar:7.6.0 (present, but unavailable): 
Could not find artifact io.confluent:kafka-schema-registry-client:jar:7.6.0 in splunk-artifactory (https://splunk.jfrog.io/splunk/ext-releases-local)
```

<details><summary>expanded output</summary>

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Google Cloud Dataflow Templates 1.0-SNAPSHOT:
[INFO] 
[INFO] Google Cloud Dataflow Templates .................... SUCCESS [  0.137 s]
[INFO] Templates Metadata Module .......................... SUCCESS [  0.783 s]
[INFO] Templates Plugins Project .......................... SUCCESS [  0.003 s]
[INFO] Templates Core Plugin .............................. SUCCESS [  0.588 s]
[INFO] Templates Maven Plugin ............................. SUCCESS [  2.010 s]
[INFO] integration-testing-lib ............................ SUCCESS [  0.094 s]
[INFO] it-google-cloud-platform ........................... SUCCESS [  1.518 s]
[INFO] it-datadog ......................................... SUCCESS [  0.565 s]
[INFO] it-conditions ...................................... SUCCESS [  0.235 s]
[INFO] it-jdbc ............................................ SUCCESS [  0.362 s]
[INFO] it-splunk .......................................... SUCCESS [  0.345 s]
[INFO] Google Cloud Teleport Classic ...................... SUCCESS [  9.109 s]
[INFO] structured-logging ................................. SUCCESS [  0.070 s]
[INFO] plaintext-logging .................................. SUCCESS [  0.005 s]
[INFO] Google Cloud Teleport Dynamic ...................... SUCCESS [  0.778 s]
[INFO] common ............................................. SUCCESS [  2.082 s]
[INFO] AstraDB To BigQuery ................................ SUCCESS [  2.296 s]
[INFO] kafka-common ....................................... FAILURE [  2.594 s]
[INFO] azure-eventhub-to-pubsub ........................... SKIPPED
[INFO] bigtable-common .................................... SKIPPED
[INFO] bigtable-changestreams-to-hbase .................... SKIPPED
[INFO] bigquery-to-bigtable ............................... SKIPPED
[INFO] bigquery-to-parquet ................................ SKIPPED
[INFO] cdc-parent ......................................... SKIPPED
[INFO] cdc-common ......................................... SKIPPED
[INFO] cdc-embedded-connector ............................. SKIPPED
[INFO] cdc-change-applier ................................. SKIPPED
[INFO] cdc-agg ............................................ SKIPPED
[INFO] jdbc-common ........................................ SKIPPED
[INFO] dataplex ........................................... SKIPPED
[INFO] datastream-common .................................. SKIPPED
[INFO] datastream-to-bigquery ............................. SKIPPED
[INFO] datastream-to-mongodb .............................. SKIPPED
[INFO] datastream-to-postgres ............................. SKIPPED
[INFO] spanner-migrations-sdk ............................. SKIPPED
[INFO] spanner-common ..................................... SKIPPED
[INFO] datastream-to-spanner .............................. SKIPPED
[INFO] datastream-to-sql .................................. SKIPPED
[INFO] elasticsearch-common ............................... SKIPPED
[INFO] file-format-conversion ............................. SKIPPED
[INFO] gcs-to-sourcedb .................................... SKIPPED
[INFO] googlecloud-to-elasticsearch ....................... SKIPPED
[INFO] googlecloud-to-googlecloud ......................... SKIPPED
[INFO] Google Cloud To MongoDB ............................ SKIPPED
[INFO] googlecloud-to-neo4j ............................... SKIPPED
[INFO] googlecloud-to-splunk .............................. SKIPPED
[INFO] google-ads-to-googlecloud .......................... SKIPPED
[INFO] jdbc-to-googlecloud ................................ SKIPPED
[INFO] jms-to-pubsub ...................................... SKIPPED
[INFO] kafka-to-bigquery .................................. SKIPPED
[INFO] kafka-to-gcs ....................................... SKIPPED
[INFO] kafka-to-kafka ..................................... SKIPPED
[INFO] kafka-to-pubsub .................................... SKIPPED
[INFO] kinesis-to-pubsub .................................. SKIPPED
[INFO] kudu-to-bigquery ................................... SKIPPED
[INFO] MongoDB To Googlecloud ............................. SKIPPED
[INFO] mqtt-to-pubsub ..................................... SKIPPED
[INFO] mysql-to-googlecloud ............................... SKIPPED
[INFO] oracle-to-googlecloud .............................. SKIPPED
[INFO] postgresql-to-googlecloud .......................... SKIPPED
[INFO] pubsub-binary-to-bigquery .......................... SKIPPED
[INFO] pubsub-cdc-to-bigquery ............................. SKIPPED
[INFO] pubsub-to-jms ...................................... SKIPPED
[INFO] pubsub-to-kafka .................................... SKIPPED
[INFO] PubSub To MongoDB .................................. SKIPPED
[INFO] PubSub To Redis .................................... SKIPPED
[INFO] sourcedb-to-spanner ................................ SKIPPED
[INFO] spanner-change-streams-to-sharded-file-sink ........ SKIPPED
[INFO] spanner-custom-shard ............................... SKIPPED
[INFO] sqlserver-to-googlecloud ........................... SKIPPED
[INFO] streaming-data-generator ........................... SKIPPED
[INFO] Templates Python Wrapper Module .................... SKIPPED
[INFO] Templates Yaml Wrapper Module ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24.131 s
[INFO] Finished at: 2024-07-02T15:14:03-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (enforce) on project kafka-common: 
[ERROR] Rule 0: org.codehaus.mojo.extraenforcer.dependencies.EnforceBytecodeVersion failed with message:
[ERROR] The following artifacts could not be resolved: io.confluent:kafka-schema-registry-client:jar:7.6.0 (present, but unavailable), io.confluent:kafka-avro-serializer:jar:7.6.0 (present, but unavailable), io.confluent:common-utils:jar:7.6.0 (present, but unavailable), io.confluent:kafka-schema-serializer:jar:7.6.0 (present, but unavailable): Could not find artifact io.confluent:kafka-schema-registry-client:jar:7.6.0 in splunk-artifactory (https://splunk.jfrog.io/splunk/ext-releases-local)
```

</details>

--- a/.github/actions/publish-site-report/action.yml
+++ b/.github/actions/publish-site-report/action.yml
@@ -30,7 +30,7 @@ runs:
   steps:
     - name: Generate Site Report
       shell: bash
-      run: mvn -B surefire-report:report-only -f pom.xml -Daggregate=true
+      run: mvn -B surefire-report:report-only -f pom.xml -Daggregate=true -Denforcer.skip=true
     - name: Publish Site Report
       uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v3.1.2
       with:

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -98,7 +98,7 @@ jobs:
       if: always()
     - name: Create artifacts and push
       run: |
-        mvn verify -PtemplatesRelease,splunkDeps,missing-artifact-repos \
+        mvn verify -PtemplatesRelease \
         -DprojectId="dataflow-templates" \
         -DbucketName="dataflow-templates-staging" \
         -DlibrariesBucketName="dataflow-templates-libraries" \

--- a/cicd/internal/workflows/maven-workflows.go
+++ b/cicd/internal/workflows/maven-workflows.go
@@ -106,15 +106,15 @@ func (*mvnFlags) FailAtTheEnd() string {
 }
 
 func (*mvnFlags) RunIntegrationTests() string {
-	return "-PtemplatesIntegrationTests,splunkDeps,missing-artifact-repos"
+	return "-PtemplatesIntegrationTests"
 }
 
 func (*mvnFlags) RunIntegrationSmokeTests() string {
-	return "-PtemplatesIntegrationSmokeTests,splunkDeps,missing-artifact-repos"
+	return "-PtemplatesIntegrationSmokeTests"
 }
 
 func (*mvnFlags) RunLoadTests() string {
-	return "-PtemplatesLoadTests,splunkDeps,missing-artifact-repos"
+	return "-PtemplatesLoadTests"
 }
 
 // The number of modules Maven is going to build in parallel in a multi-module project.

--- a/it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java
+++ b/it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/TemplateTestBase.java
@@ -393,7 +393,7 @@ public abstract class TemplateTestBase {
       "-pl",
       moduleBuild,
       "-am",
-      "-PtemplatesStage,pluginOutputDir,splunkDeps,missing-artifact-repos",
+      "-PtemplatesStage,pluginOutputDir",
       "-DpluginRunId=" + RandomStringUtils.randomAlphanumeric(16),
       // Skip shading for now due to flakiness / slowness in the process.
       "-DskipShade=" + skipShade,

--- a/it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/dataflow/FlexTemplateDataflowJobResourceManager.java
+++ b/it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/dataflow/FlexTemplateDataflowJobResourceManager.java
@@ -218,7 +218,7 @@ public class FlexTemplateDataflowJobResourceManager implements ResourceManager {
       "-pl",
       moduleBuild,
       "-am",
-      "-PtemplatesStage,pluginOutputDir,splunkDeps,missing-artifact-repos",
+      "-PtemplatesStage,pluginOutputDir",
       // Skip shading for now due to flakiness / slowness in the process.
       "-DskipShade=" + true,
       "-DskipTests",

--- a/it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/dataflow/FlexTemplateDataflowJobResourceManagerTest.java
+++ b/it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/dataflow/FlexTemplateDataflowJobResourceManagerTest.java
@@ -49,7 +49,7 @@ public class FlexTemplateDataflowJobResourceManagerTest {
     assertThat(String.join(" ", Arrays.copyOfRange(actual, 0, 5)))
         .isEqualTo("mvn compile package -q -f");
     String expected =
-        "-pl metadata,v2/common,v2/spanner-change-streams-to-sharded-file-sink -am -PtemplatesStage,pluginOutputDir,splunkDeps,missing-artifact-repos -DskipShade=true -DskipTests -Dmaven.test.skip -Dcheckstyle.skip -Dmdep.analyze.skip -Dspotless.check.skip -Denforcer.skip -DprojectId=testProject -Dregion=us-central1 -DbucketName=TestBucketName -DgcpTempLocation=TestBucketName -DstagePrefix=TestClassName -DtemplateName=Spanner_Change_Streams_to_Sharded_File_Sink -DunifiedWorker=true -e";
+        "-pl metadata,v2/common,v2/spanner-change-streams-to-sharded-file-sink -am -PtemplatesStage,pluginOutputDir -DskipShade=true -DskipTests -Dmaven.test.skip -Dcheckstyle.skip -Dmdep.analyze.skip -Dspotless.check.skip -Denforcer.skip -DprojectId=testProject -Dregion=us-central1 -DbucketName=TestBucketName -DgcpTempLocation=TestBucketName -DstagePrefix=TestClassName -DtemplateName=Spanner_Change_Streams_to_Sharded_File_Sink -DunifiedWorker=true -e";
     assertThat(String.join(" ", Arrays.copyOfRange(actual, 6, 25))).isEqualTo(expected);
     assertThat(actual[5]).endsWith("pom.xml");
     assertThat(actual[25]).isNotEmpty();

--- a/pom.xml
+++ b/pom.xml
@@ -391,6 +391,11 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-report-plugin</artifactId>
+        <version>2.6</version>
+      </plugin>
     </plugins>
     <extensions>
       <extension>

--- a/pom.xml
+++ b/pom.xml
@@ -392,6 +392,13 @@
         </executions>
       </plugin>
     </plugins>
+    <extensions>
+      <extension>
+        <groupId>com.google.cloud.artifactregistry</groupId>
+        <artifactId>artifactregistry-maven-wagon</artifactId>
+        <version>2.2.0</version>
+      </extension>
+    </extensions>
   </build>
 
   <profiles>
@@ -601,9 +608,31 @@
       </repositories>
     </profile>
     <profile>
+      <id>internalMaven</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+      <repositories>
+        <repository>
+          <id>central</id>
+          <name>Maven Central remote repository</name>
+          <url>artifactregistry://us-maven.pkg.dev/artifact-foundry-prod/maven-3p</url>
+          <layout>default</layout>
+          <releases>
+            <enabled>true</enabled>
+          </releases>
+          <snapshots>
+            <enabled>true</enabled>
+          </snapshots>
+        </repository>
+      </repositories>
+    </profile>
+    <profile>
       <id>splunkDeps</id>
       <activation>
-        <activeByDefault>true</activeByDefault>
+        <property>
+          <name>!skipSplunkDeps</name>
+        </property>
       </activation>
       <repositories>
         <repository>
@@ -622,6 +651,21 @@
           <url>https://oss.sonatype.org/content/repositories/snapshots</url>
         </pluginRepository>
       </pluginRepositories>
+    </profile>
+    <profile>
+      <id>confluentDeps</id>
+      <activation>
+        <property>
+          <name>!skipConfluentDeps</name>
+        </property>
+      </activation>
+      <repositories>
+        <repository>
+          <!-- Required for io.confluent:kafka-avro-serializer artifact -->
+          <id>confluent</id>
+          <url>https://packages.confluent.io/maven/</url>
+        </repository>
+      </repositories>
     </profile>
     <profile>
       <!-- mvn clean deploy -Prelease -pl plugins/templates-maven-plugin -pl v2/common -am -->

--- a/v1/pom.xml
+++ b/v1/pom.xml
@@ -949,19 +949,6 @@
 
   <profiles>
     <profile>
-      <id>missing-artifact-repos</id>
-      <activation>
-        <activeByDefault>true</activeByDefault>
-      </activation>
-      <repositories>
-        <repository>
-          <!-- Required for io.confluent:kafka-avro-serializer artifact -->
-          <id>confluent</id>
-          <url>https://packages.confluent.io/maven/</url>
-        </repository>
-      </repositories>
-    </profile>
-    <profile>
       <id>shade</id>
       <activation>
         <property>

--- a/v2/pom.xml
+++ b/v2/pom.xml
@@ -468,19 +468,6 @@
 
     <profiles>
         <profile>
-            <id>missing-artifact-repos</id>
-            <activation>
-                <activeByDefault>true</activeByDefault>
-            </activation>
-            <repositories>
-                <repository>
-                    <!-- Required for io.confluent:kafka-avro-serializer artifact -->
-                    <id>confluent</id>
-                    <url>https://packages.confluent.io/maven/</url>
-                </repository>
-            </repositories>
-        </profile>
-        <profile>
             <id>shade</id>
             <activation>
                 <property>

